src/app/features/asset-list/stacks/stacks-assets-list/stacks-assets-list.tsx

```diff
@@ -0,0 +1,98 @@
+import { type ReactNode } from 'react';
+
+import type { Sip10CryptoAssetFilter } from '@leather.io/query';
+import { StxAvatarIcon } from '@leather.io/ui';
+
+import { CryptoAssetItemLayout } from '@app/components/crypto-asset-item/crypto-asset-item.layout';
+import { Sip10TokensLoader } from '@app/components/loaders/sip10-tokens-loader';
+import { CurrentStacksAccountLoader } from '@app/components/loaders/stacks-account-loader';
+import { Stx20TokensLoader } from '@app/components/loaders/stx20-tokens-loader';
+import { StxAssetItemBalanceLoader } from '@app/components/loaders/stx-balance-loader';
+
+import { ConnectLedgerAssetItemFallback } from '../../_components/connect-ledger-asset-item-fallback';
+import type { AssetListVariant } from '../../asset-list';
+import { Sip10TokenAssetList } from '../sip10-token-asset-list/sip10-token-asset-list';
+import { Stx20TokenAssetList } from '../stx20-token-asset-list/stx20-token-asset-list';
+import { StxCryptoAssetItem } from '../stx-crypo-asset-item/stx-crypto-asset-item';
+
+interface StacksAssetsListProps {
+  isReadOnly: boolean;
+  variant: AssetListVariant;
+  onSelectAsset?(symbol: string, contractId?: string): void;
+  renderRightElement?(id: string): ReactNode;
+  filter?: Sip10CryptoAssetFilter;
+  showBalance?: boolean;
+  showStx?: boolean;
+  hasTokenSetter?: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export function StacksAssetsList({
+  isReadOnly,
+  variant,
+  onSelectAsset,
+  renderRightElement,
+  filter,
+  hasTokenSetter,
+  showBalance = true,
+  showStx = true,
+}: StacksAssetsListProps) {
+  return (
+    <CurrentStacksAccountLoader
+      fallback={
+        <ConnectLedgerAssetItemFallback
+          chain="stacks"
+          icon={<StxAvatarIcon />}
+          symbol="STX"
+          variant={variant}
+        />
+      }
+    >
+      {account => (
+        <>
+          {showStx && filter != 'unsupported' ? (
```

Comment:
@cbadawi can you please refactor this to not use nested ternary statements? 

Nesting ternary statements makes the code hard to read and the logic tricky to follow